File: nvpl_blas/f77/srotg.f
!******************************************************************************
! Content:
!     This example demonstrates use of API as below:
!     SROTG 
!
!******************************************************************************
      program   SROTG_MAIN
*
      implicit none

      include "nvpl_blas_f77_blas.fi"

      real            a, b, c, s
      data            a /2.3/, b /4.6/, c /0.0/, s /0.0/
      

*
*      Executable Statements
*
      print*
      print 99
*
*       Print input data
      print*
      print 100, a, b, c, s
*
*      Call SROTG subroutine
      call SROTG(a, b, c, s)
*
      print*
      print 101
      print 102, a, b, c, s

      stop
 99   format('Example: SROTG for computing the parameters for a Givens'
     & ' rotation')
 100  format('#### args: a=',f5.3,', b=',f5.3,', c=',f5.3,', '
     &       's=',f5.3)
 101  format('The result of SROTG: ')
 102  format('a=',f5.3,', b=',f5.3,', c=',f5.3,', s=',f5.3)
      end

File: nvpl_blas/c/srotg.c
/******************************************************************************
 * Content: 
 *     This example demonstrates use of API as below:
 *     cblas_srotg
 *
 ******************************************************************************/
#include "example_helper.h"

int main() {
    float a = 2.3f;
    float b = 4.6f;
    float c = 0.0f;
    float s = 0.0f;

    printf("\nExample: cblas_srotg for computing the parameters for a Givens rotation\n");
    printf("\n#### args: a=%f, b=%f, c=%f, s=%f\n",a, b, c, s);

    // call cblas_srotg
    cblas_srotg(&a, &b, &s, &c);

    // print result
    printf("\nThe result of cblas_srotg:\n");
    printf("a=%f, b=%f, c=%f, s=%f\n",a, b, c, s);

    return 0;
}
